examples/fabric-sync/shell/ShellCommands.cpp

```diff
@@ -144,6 +146,36 @@ static CHIP_ERROR HandleAddDeviceCommand(int argc, char ** argv)
     return result;
 }
 
+static CHIP_ERROR HandlePairDeviceCommand(int argc, char ** argv)
+{
+    if (argc != 3)
+    {
+        fprintf(stderr, "Invalid arguments. Usage: app pair-device node-id code\n");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // Check if there is already an active command
+    if (commands::CommandRegistry::Instance().IsCommandActive())
+    {
+        fprintf(stderr, "Another command is currently active. Please wait until it completes.\n");
+        return CHIP_ERROR_BUSY;
+    }
+
+    // Parse arguments
+    chip::NodeId nodeId    = static_cast<chip::NodeId>(strtoull(argv[1], nullptr, 10));
+    const char * setUpCode = argv[2];
+
+    auto command = std::make_unique<commands::PairDeviceCommand>(nodeId, setUpCode);
```

Comment:
After this, the call is pretty much async right? How do you know that setUpCode's memory doesn't change between when this is called and when it is used by PairDeviceCommand?

Reply:
It is synchronous until:

return admin::PairingManager::Instance().PairDeviceWithCode(mNodeId, mPayload);

The we need to make a copy of mPayload within PairingManager:PairDeviceWithCode to its instance buffer and process the command asynchronously.

Reply:
I don't think that is correct:

`commands::CommandRegistry::Instance().SetActiveCommand` literally schedules the task to run so everything about that command is async

https://github.com/project-chip/connectedhomeip/blob/master/examples/fabric-sync/shell/CommandRegistry.cpp#L27-L40

Reply:
Oh.... nvm I mis-understood something I am review again, sorry

Reply:
It all looks good. My mistake. Thanks for explaining